__tests__/fetcher/fetch-page-data.ts

```diff
@@ -23,3540 +37,491 @@ afterAll(() => {
-              },
-            },
-          },
-        })
-      )
+describe('fetcher: check examples for all supported typenames', () => {
+  test('typename: User', async () => {
```

Comment:
Better: 

```typescript
  describe("returns an error for unsupported types", () => {
    test('returns an error when a user is requested because ...', async () => {
      ...
    }
  }
```

The reason helps me to understand why the return type is not `UserPage` rather than `ErrorPage`. So you can use the tests for documenting the current state of the project.